src/hoi4_world/states/hoi4_states_converter.cpp

```diff
@@ -76,6 +76,73 @@ std::optional<int> GetStateIfSharedByAllProvinces(const std::vector<std::string>
 }
 
 
+std::optional<int> FindStateWithMostOfSignificantProvinceType(const std::string& significant_province_type,
```

Comment:
How can a state have more than one city/farm/mine/port? Isn't this FindStateWithSignificantProvinceType? I guess the implementation is future proofed, but the name threw me off for a bit there.

Reply:
It's more likely a reverse. There's states without ports for example, by accident or deliberate. Likely there's states where they forgot to add other significant province types.

Reply:
A little of both. In practice there should only be one, but that 'should' worries me.